### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by Gerych1984**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

1. More tests
2. Small fixes
3. Allow `DateTimeInterface` as values
4. DataReader and Processors using `QueryInterface` instead of `Query` where is possible
5. Add filter `IsNull`
6. Rename `prepareQuery` to `getPreparedQuery` and change access to public. It can be useful for add some not standart conditions, AR relations, show result raw query string and explain it
7. All processors now final

### CODE REVIEW

**Review by vjik — APPROVED** [review]

**Comment by samdark** [comment]
Would you please resolve conflicts?

**Comment by Gerych1984** [comment]
> Would you please resolve conflicts?

Done

File: src/Filter/CompareFilter.php [hunk 1]
```diff
@@ -4,26 +4,23 @@
...
 
 abstract class CompareFilter implements FilterInterface
 {
-    /**
-     * @var ExpressionInterface|string
-     */
-    protected $column;
+    use ColumnFormatterTrait;
+
+    public static string $mainDateTimeFormat = 'Y-m-d H:i:s';
```
Comment (samdark) [inline src/Filter/CompareFilter.php:15]: Is that meant to be a constant?

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

File: src/Filter/CompareFilter.php (hunk 1 above)
Comment (Gerych1984) [reply src/Filter/CompareFilter.php:15]: > Is that meant to be a constant?

No - that default format can be changed by user for all instances (in bootstrap.php). By default it most common format, but in my case i using pgsql timestamp_tz

**Pull request merged**